packages/analytics-core/src/plugins/destination.ts

```diff
@@ -297,21 +298,33 @@ export class Destination implements DestinationPlugin {
   }
 
   fulfillRequest(list: Context[], code: number, message: string) {
-    this.saveEvents();
     list.forEach((context) => context.callback(buildResult(context.event, code, message)));
+    void this.updateEventStorage(list);
   }
 
   /**
-   * Saves events to storage
-   * This is called on
-   * 1) new events are added to queue; or
-   * 2) response comes back for a request
+   * update the event storage
    */
-  saveEvents() {
+  async updateEventStorage(filterList: Context[], newContext?: Context[]) {
```

Comment:
nit: given the similarity of these params, and both are optional based on usage, it could be could to make these named to avoid accidentally setting the wrong value and prevent needing to pass in an empty filter list like on line 105.

```suggestion
  async updateEventStorage({ filterList?: Context[], newContext?: Context[] }) {
```

Reply:
Sync offline, updated the parameters to eventsToRemove and eventsToAdd instead of making the input an object.